RecoVertex/PrimaryVertexProducer/interface/DAClusterizerInZT.h

```diff
@@ -0,0 +1,108 @@
+#ifndef DAClusterizerInZT_h
+#define DAClusterizerInZT_h
+
+/**\class DAClusterizerInZT 
+ 
+  Description: separates event tracks into clusters along the beam line
+
+*/
+
+#include "RecoVertex/PrimaryVertexProducer/interface/TrackClusterizerInZ.h"
+#include "TrackingTools/TransientTrack/interface/TransientTrack.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include <vector>
+#include "DataFormats/Math/interface/Error.h"
+#include "RecoVertex/VertexTools/interface/VertexDistanceXY.h"
+#include "RecoVertex/VertexPrimitives/interface/TransientVertex.h"
+
+
+
+
+class DAClusterizerInZT : public TrackClusterizerInZ {
+
+public:
+
+struct track_t{
+  double z;              // z-coordinate at point of closest approach to the beamline
+  double t;              // t-coordinate at point of closest approach to the beamline
+  double dz2;            // square of the error of z(pca)
+  double dtz;            // covariance of z-t
+  double dt2;            // square of the error of t(pca)
+  const reco::TransientTrack* tt;  // a pointer to the Transient Track
+  double Z;              // Z[i]   for DA clustering
```

Comment:
Z vs z doesn't look like a good choice for development.
The capital Z is also against CMS style for variable names.